### PR TITLE
Talk about recipient limits in spec

### PR DESCRIPTION
**Pull request by akalin-keybase**
Correct recipient count check -- it should be
(2^32) - 1 and not 2^31.

### CODE REVIEW

File: encrypt.go [hunk 1]
```diff
@@ -163,8 +163,8 @@ func (es *encryptStream) checkReceivers(v []BoxPublicKey) error {
 		tot++
 	}
 
-	// Don't allow more than 2^31 receivers.
-	if tot >= 0x7fffffff {
+	// Don't allow more than 2^32-1 receivers.
+	if tot > ((1 << 32) - 1) {
```
Comment (oconnor663) [inline encrypt.go]: Is this correct on 32-bit systems?

File: specs/saltpack_encryption_v1.md [hunk 2]
```diff
@@ -95,8 +95,12 @@ The header packet is a MessagePack list with these contents:
 - The **sender secretbox** is a
   [`crypto_secretbox`](http://nacl.cr.yp.to/secretbox.html) containing the
   sender's long-term public key, encrypted with the **payload key** from below.
-- The **recipients list** contains a recipient pair for each recipient key,
-  including an encrypted copy of the **payload key**. See below.
+- The **recipients list** contains a recipient pair for each recipient
+  key, including an encrypted copy of the **payload key** (see
+  below). Note that a MessagePack array can hold
+  [at most 2³² &minus; 1](https://github.com/msgpack/msgpack/blob/master/spec.md#array-format-family)
```
Comment (oconnor663) [inline specs/saltpack_encryption_v1.md]: Unicode shenanigans :-D

Were you able to test rendering this to make sure it looks ok?

File: encrypt.go (hunk 1 above)
Comment (akalin-keybase) [reply encrypt.go]: Yeap, https://golang.org/ref/spec#Constants says constants are calculated with at least 256-bit precision, and we only compare it to a `uint64`. Tested with `GOARCH=i386` to make sure.

File: specs/saltpack_encryption_v1.md (hunk 2 above)
Comment (akalin-keybase) [reply specs/saltpack_encryption_v1.md]: Yeah, I did so by modifying the HTML on the saltpack webpage with Chrome inspector.

**Pull request merged**